tests/test_xjoin.py

```diff
@@ -1684,6 +1684,51 @@ def test_query_hosts_filter_spf_insights_client_version(
                 graphql_query_empty_response.reset_mock()
 
 
+# system_profile insights_client_version tests
+def test_query_hosts_filter_spf_operating_system(
+    mocker, subtests, query_source_xjoin, graphql_query_empty_response, patch_xjoin_post, api_get
+):
+    http_queries = (
+        "filter[system_profile][operating_system][major]=7",
+        "filter[system_profile][operating_system][minor]=3",
+        "filter[system_profile][operating_system][name]=RHEL",
+        (
+            "filter[system_profile][operating_system][major]=7&"
+            "filter[system_profile][operating_system][minor]=3&"
+            "filter[system_profile][operating_system][name]=RHEL"
+        ),
+    )
+
+    graphql_queries = (
+        {"spf_operating_system": {"major": {"eq": "7"}}},
+        {"spf_operating_system": {"minor": {"eq": "3"}}},
+        {"spf_operating_system": {"name": {"eq": "RHEL"}}},
+        {"spf_operating_system": {"major": {"eq": "7"}, "minor": {"eq": "3"}, "name": {"eq": "RHEL"}}},
+    )
+
+    for http_query, graphql_query in zip(http_queries, graphql_queries):
+        with subtests.test(http_query=http_query, graphql_query=graphql_query):
+            url = build_hosts_url(query=f"?{http_query}")
+
+            response_status, response_data = api_get(url)
```

Comment:
Since `response_data` isn't used, would it make more sense to do it this way?
```suggestion
            response_status = api_get(url)[0]
```

Reply:
:+1: